users/users_get_test.go

```diff
@@ -17,3 +23,317 @@ func (u *User) bindExisting(ctx context.Context, tb testing.TB, userID string) *
 
 	return u
 }
+
+//nolint:funlen //long is nice
+func TestRepository_GetUsers_Success_Common(t *testing.T) { //nolint:paralleltest // We need a clean database.
+	if testing.Short() {
+		return
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), testDeadline)
+	defer cancel()
+	var usrs []*User
+	var res []*RelatableUserProfile
+	var err error
+	const userCount = 10
+	SETUP("we cleanup everything in the database", func() { mustDeleteAllUsers(ctx, t) })
+	GIVEN("create all users", func() {
+		usrs = mustCreateMutuallyReferringUsers(ctx, t, userCount)
+		ctx = context.WithValue(ctx, "requestingUserIDCtxValueKey", usrs[0].ID) //nolint:staticcheck // No way
+	})
+	for _, keyword := range []string{"User", "First", "Last"} {
+		for _, limit := range []uint64{0, 5, 100} {
+			for _, offset := range []uint64{0, 5} {
+				WHEN("get all users", func() {
+					res, err = usersRepository.GetUsers(ctx, keyword, limit, offset)
+				})
+				THEN(func() {
+					IT("is it ok", func() {
+						require.NoError(t, err)
+					})
+					IT("check number of elements <= limit", func() {
+						assert.LessOrEqual(t, len(res), int(limit))
```

Comment:
```suggestion
						assert.LessOrEqual(t, int(limit), len(res))
```
**expected** value is the **second** argument, **actual** is the **third**. In this case `res` is actual.

Reply:
done